### PR TITLE
initial support

### LINKED ISSUE #554: Support for Go Modules
Example file: https://github.com/mattermost/mattermost-server/blob/master/go.mod

### PR DESCRIPTION
**Pull request by StarpTech**
Fix #554 

Provides initial support for https://golang.org/cmd/go/#hdr-The_go_mod_file

Tested with: https://github.com/mattermost/mattermost-server/blob/master/go.mod

### Checklist:

- [X] If this PR is a new feature, please provide at least one example link
- [x] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

File: packages/helper-grammar-regex-collection/test.js [hunk 1]
```diff
@@ -387,12 +387,33 @@ const fixtures = {
...
     ],
     invalid: [
       'simport foo',
       'simport\nfoo',
       'import "octo.com/foo/bar"',
       'import (\n"octo.com/foo/bar"\n)',
+
+      'require (github.com/foo v1.1.0)',
+      'require \n"foo" v1.1.0',
+      'srequire "foo" v1.1.0',
```
Comment (xt0rted) [inline packages/helper-grammar-regex-collection/test.js]: @stefanbuck looks like the npm parser is picking these two lines up.

image

**Comment by StarpTech** [comment]
No idea what is causing the E2E to fail.

File: packages/helper-grammar-regex-collection/test.js (hunk 1 above)
Comment (StarpTech) [reply packages/helper-grammar-regex-collection/test.js]: How do you know?

Comment (xt0rted) [reply packages/helper-grammar-regex-collection/test.js]: It's picking up `foo` and linking it to https://github.com/jasonLaster/foo. It's not an issue with this PR, that's something with the current npm parser that needs to be looked at.

Comment (stefanbuck) [reply packages/helper-grammar-regex-collection/test.js]: @xt0rted good spot, but I wonder if its worth a fix. The file extension is `.js` and therefore the `require` regexp matches this string as well.

**Comment by stefanbuck** [comment]
@StarpTech thanks for taking the time to draft and submit this PR.


I'll have a closer look tomorrow, but just one thing upfront. It would be really amazing if you could add an E2E test to cover this new behaviour. It's super simple see https://github.com/OctoLinker/OctoLinker/tree/master/e2e 

Regarding failing E2E tests, this happens from time to time and often a re-run helps.

**Comment by stefanbuck** [comment]
I re-run the CI job, but it seems to stuck on the E2E tests for an unknown reason.

**Comment by stefanbuck** [comment]
The job failed now. I don't think it's caused by the changes you made. 'll look into it.

**Comment by StarpTech** [comment]
Hi, @stefanbuck have you looked into?

**Comment by stefanbuck** [comment]
I did, but I can't remember the outcome 😄 I'll have another look tonight, sorry for the delay on this

**Comment by stefanbuck** [comment]
Debug log:

- I rebased and pushed the changes to your fork -> CI failed
- I removed all your changes and replace it with a dummy change -> CI still fails
- I pushed your changes to my remote and opened a new PR -> CI passed

This is super weird and I have no idea what is causing this behaviour. Maybe best to get your changes in as part of my PR https://github.com/OctoLinker/OctoLinker/pull/879. You're still the original commit author so your contribution is not lost 😉 

I'll merge your change later next week, because master is blocked by another release which needs to get out first next week.

**Comment by stefanbuck** [comment]
Ohh just one thing, did you looked at the E2E tests? To ensure this feature continues to work, I would prefer to have them in place. Doesn't have to be crazy and as mention those are super easy to add. I can cherry-pick them from this branch or feel free to push to mine whatever works best for you.

**Comment by stefanbuck** [comment]
Closing this in favour #879

**Pull request closed without merging**